### ISSUE TITLE
ENH: Adding ITAB file reader

### ISSUE THREAD

**Issue by robbisg** [mne-tools/mne-python#12439]
### Describe the new feature or enhancement

We would like to add the ITAB raw file reader to MNE-Python. @vpizzella made a first draft of the reader and I am prepping it following your guidelines with tests and linting.
I have a bunch of question regarding the management of testing data and tests:
1) I noticed that data is included in another repo, should I open a PR also in this repo?
2) Is there a size limit of the files? 
3) Is it better a file that includes all features implemented (and tested) or it is better a set of files?
4) I used a matlab (fieldtrip) file to test whether the reader reads the same numbers, is it correct or is there any other strategy?

### Describe your proposed implementation

A first draft of the implementation can be found [here](https://github.com/mambolab/mne-python/tree/itab_reader), but it needs linting and so on.

### Describe possible alternatives

We followed other readers implementation, so it is similar to other readers and maybe optimal.

**Comment by larsoner** [comment]
> I noticed that data is included in another repo, should I open a PR also in this repo?
> Is there a size limit of the files?

Yep, PR there first is great. "As small as possible" is the rule of thumb. In practice for most formats we get away with something < 1 MB. Basically ~1sec of data or less is hopefully enough. But if your acquisition system writes like 10s blocks or something, then you'd want 20s of data for example. 

> Is it better a file that includes all features implemented (and tested) or it is better a set of files?

I would say whatever uses the least disk space while providing proper code/functionality coverage. So for example a file with weird lowpass settings and UTF encoding, and another file with normal lowpass settings and latin-1 encoding would be fine and preferred over 4 files (normal lowpass, weird lowpass, utf8, latin-1).

> I used a matlab (fieldtrip) file to test whether the reader reads the same numbers, is it correct or is there any other strategy?

Yes if you have a reference implementation you know/trust is correct then often what we do is have the native format plus a little `.mat` file containing the data read with FieldTrip or EEGLAB or whatever, then we `assert_allclose(raw.get_data(), read_mat(...)['data'])` (paraphrasing).

**Comment by HansujaB** [comment]
@larsoner I went through the referred implementation above ,  I am up to work on it further. I looked for a file we could add to testing data but I couldn't find  it , I believe they are not released as such. What do you suggest?

**Comment by larsoner** [comment]
I think we have to wait for someone to provide a suitable test file ☹️

**Comment by HansujaB** [comment]
Hi , @robbisg  do you have a suitable file that we can use?

**Comment by robbisg** [comment]
Sorry for the late reply. 
We have a first implementation of the reader that we used internally, but we didn't extensively test the reader. If you want in the next days, I can open a PR with this feature and maybe send a link to a test file so that if you are interested in completing the implementation you can go on.
Thanks!

**Comment by HansujaB** [comment]
@robbisg  Thanks for getting back! That would be really helpful.
If you can open a PR with the current implementation and share a sample test file, I’d be happy to pick it up from there and work on improving the reader and adding proper test coverage.
